sdkproject/Assets/Mapbox/Unity/MeshGeneration/Data/UnityTile.cs

```diff
@@ -121,6 +121,11 @@ internal void Recycle()
 			HeightDataState = TilePropertyState.None;
 			VectorDataState = TilePropertyState.None;
 
+			OnHeightDataChanged = null;
```

Comment:
it was supposed to be `delegate { };` , not null to clear events when the tile is recycled, to prevent events from previous tile of course.